File: app/src/main/java/com/tamil/smshelper/data/MessageTemplateDao.java
package com.tamil.smshelper.data;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface MessageTemplateDao {

    @Query("SELECT * FROM message_template ORDER BY id ASC")
    List<MessageTemplate> getAllTemplates();

    @Insert
    void insertAll(MessageTemplate... templates);

    @Update
    void updateTemplate(MessageTemplate template);

    @Query("SELECT * FROM message_template WHERE isSelected = 1 LIMIT 1")
    MessageTemplate getSelectedTemplate();

    @Query("UPDATE message_template SET isSelected = 0")
    void clearAllSelections();
}

File: app/src/main/java/com/tamil/smshelper/data/MessageTemplateDao.kt
package com.tamil.smshelper.data

import androidx.room.*

@Dao
interface MessageTemplateDao {

    @Query("SELECT * FROM message_templates ORDER BY id ASC")
    fun getAllTemplates(): List<MessageTemplate>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(templates: List<MessageTemplate>)

    @Update
    fun updateTemplate(template: MessageTemplate)

    @Query("UPDATE message_templates SET isSelected = 0")
    fun clearAllSelections()

    @Query("SELECT * FROM message_templates WHERE isSelected = 1 LIMIT 1")
    fun getSelectedTemplate(): MessageTemplate?
}
